[HBHE] Update MahiDebugger and fix nP in NNLS

jaehyeok opened the pull request:
This PR consists of two changes.

(1) update on MahiDebugger  

The current version of it only stores info for BX=-1,0,1 pulses and baseline because mahi ran with 3 pulses + baseline. Now it runs with 8 pulses, so the debugger should be updated accordingly. This has no impact on the reconstruction because changes are only in the debugger.

(2) Bug fix in initializing nP variable
 
Currently, nnlsWork_.nP is reset to 0 at every iteration. This turned out to make the fit unstable sometimes (particularly, the BX=+1 pulse became anomalously large). The initialization is now done at the beginning of MahiFit::phase1Apply. The comparison before and after this update is shown in [1] (Fix #2 in the slides corresponds to this update). 

Here I summarize the changes after the update: 

* Overall impact on intime energy is small (p4)
* There are changes in BX=-1 and BX=+1 pulses (p5 and p6, respectively). Based on the pulse displays (p21-32), the fit looks much better (BX=+1 pulse looks much more reasonable).
* the chi2 became smaller (p12 and 13)

All these results have led us to conclude that the fix improves the behavior of the fit. This PR is for Run3.

[1] https://indico.cern.ch/event/855229/contributions/3597588/attachments/1924959/3188179/Hcal_mahi.pdf

schneiml committed:
Redesign one::DQM* stuff.

Rebase: changed to the old *Produce implementations.

schneiml committed:
Fix all the obvious errors

schneiml committed:
Change One* headers and base classes.

Still does not work, need do also rename some methods.

schneiml committed:
Add ParameterSet, some modules need it

schneiml committed:
Rename endRun to dqmEndRun.

Includes manual changes to two modules that override endRunProduce, for
whatever reasons.

schneiml committed:
Rename endLuminosityBlock.

schneiml committed:
Rename the remaining cases.

schneiml committed:
A bunch of manual renaming changes.

schneiml committed:
More HCAL things

schneiml committed:
A few more renames.

schneiml committed:
Last round of fixes.

steggema committed:
Add first version of TICLCandidate

steggema committed:
Add initial versions of Trackster->TICLCandidate->PF plugins

steggema committed:
Working version of TICL -> PF

steggema committed:
Add validation for PFCandidates produced via HGCal->TICL

steggema committed:
Add option for trackster energy to be taken from regression

steggema committed:
Filename consistency and other minor changes to complete TICL->PF workflow

steggema committed:
Add track plugin to TICL candidate producer

steggema committed:
Change track reference to edm::Ptr in TICLCandidate; add charge to TICL PF validation; and add a bit of documentation

steggema committed:
Refactor loops in trackster->ticl candidate plugins, and code format

steggema committed:
Move validation from official sequences for now to ticl_iterations (to be re-added later w/ modifier); avoid setting charge for neutrals

steggema committed:
Merge commit 'refs/pull/27917/head' of https://github.com/cms-sw/cmssw into from-CMSSW_11_0_0_pre9

schneiml committed:
Remove test/example core that demonstrates no-longer-supported features.

schneiml committed:
Remove Reference histograms from DQM.

The only direct usages in SiStrip and Pixel phase0 DQM where actually
never called, so it is probably save to remove them. We don't load
references in production jobs for a while now, and the references used
before where massively outdated.

They main usage was Chi2 and Kologrov quality tests. One main user was
Muon DQM, however they switched (?) to an implementation outside CMSSW
a while ago ("AutoDQM"), so I assume the exisiting code did not work
correctly any more. Other users are HLT and L1T DQM.

We will not support reference histograms inside CMSSW any further, for
the future we might provide a different method of comparing to
references as a "stand-alone" module in CMSSW harvesting (without
special functionality in DQMStore).

In current practise, comparisons to reference histograms are done in
external tools such as DQMGUI, RelMon, or AutoDQM.

schneiml committed:
Fix more usages of the Kolmogorov QTest.

schneiml committed:
Code checks

schneiml committed:
Remove unused variables

schneiml committed:
Remove more mentions of Comp2Ref QTests.

schneiml committed:
Fix DQMEDAnalyzer.h

and some bugs in DQMOneEDAnalyzer.h

schneiml committed:
Rename endRun to dqmEndRun.

Include code-format and code-checks, which caused a bit off fallout.

schneiml committed:
Correctly label the tokens.

Whatever we had before wass pretty garbage, the lumi token was not even
initialized.

The run and lumi products need to have distinct labels; @Dr15Jones IIRC
said at some point this should not be required, but EDM is pretty clear
about it:

```
Duplicate Product Identifier
...
Note that if the four parts of the branch name are the same,
then this error will occur even if the branch types differ!
```

So I use the names that we have in the "New DQMStore" prototype.

Dr15Jones committed:
Added RunInfoTestESProducer used for testing

This also included a test for RunInfoTestESProducer.

smorovic committed:
*change some FileBroker parameter types to tracked for always used parameters in
HLT configuration. file broker host will also be required in the future.
*remove old verifyAdler32 option. Use common verify checksum which will select
algorithm depending on the input event version.
*DAQ test module for DQM histograms
*added JsonMonitoring and new DQM test module to HLT test scripts
*removed obsolete and duplicated test scripts
*Unit test running fake BU and simple FU menu (test for BUFU file-based
protocol). Test variant with FRD file header also added to the script
* code-check-all for a test service suggests using std::make_shared. taking also the
opportunity to update this code to use std::thread and other std classes
replacing boost

Dr15Jones committed:
Added MagFieldConfigTestESProducer

This is used to test VolumeBasedMagneticFieldESProducerFromDB.

Dr15Jones committed:
Use mayConsumes in VolumeBasedMagneticFieldESProducerFromDB

The module decides which geometry to read from the database by
reading the value stored in the MagFieldConfig EventSetup object.
The decision cannot be made at module configuration time, only at
run time. Therefore mayConsumes must be used.

Unit tests were added to guarantee the module's behavior is the
same after this change.

steggema committed:
Add minimum probability cut for tracksters to become TICLCandidates

steggema committed:
Merge remote-tracking branch 'my-cmssw/TICLtoPF' into TICLtoPF

steggema committed:
Make particle ID probability lists and accessors consistent

steggema committed:
Address code checks

steggema committed:
Run TICL->PF on all candidates produced; allow 'unknown' as output of TICL; and fix in PDG assignment

steggema committed:
Move headers only used by plugins to plugins folder

hqucms committed:
Run inner/outer conv in batch.

tocheng committed:
Add tau pog tags

peruzzim committed:
Merge pull request #417 from cms-nanoAOD/pr_nanov6_part2_master

Test PR NanoAODv6 master [part2]

peruzzim committed:
Add JME era modifiers for 2016 and 2017

peruzzim committed:
Switch modifier logic to split year and specific dataset conditions

peruzzim committed:
Merge branch 'pr_nanov6_part2_master' into master-cmsswmaster

steggema committed:
Address review comments

Dr15Jones committed:
Removed unit tests from MagneticField/GeomBuilder

Dr15Jones committed:
Removed unused runInfo_ member data

Dr15Jones committed:
Added back LogInfo messages to VolumeBasedMagneticFieldESProducerFromDB

smorovic committed:
remove log redirection to find out why unit test fails (only) on
cmsbuild side

smorovic committed:
remove automatic unit tests for BU-FU functionality, does not pass in
cmsbuild environment, but works locally

fwyzard committed:
dumpPython(): remove wrong-type unused parameter

fwyzard committed:
PrintOptions: add parameters to the constructor

schneiml committed:
Remove overloaded endRun.

Looks like this was never called since https://github.com/cms-sw/cmssw/pull/3730 / commit cc3e009322b0

ianna committed:
Add DB reader and cleanup records

Commit:
Select channels affected by mousebite

Commit:
Code Check

makortel committed:
Use ArgumentParser, print_function

makortel committed:
Add --prune option to edmConfigDump

makortel committed:
Make Process.prune() to prune also Tasks

Martin-Grunewald committed:
Make HLTDisplacedmumumuVtxProducer a stream producer

peruzzim committed:
Merge pull request #431 from cms-nanoAOD/branch_11_0_0_pre10

Update master to 11_0_0_pre10

ianna committed:
Fix MF test

ianna committed:
code format

ianna committed:
Cleanup configuration

makortel committed:
Keep Tasks associated directly to the Schedule in prune()

fwyzard committed:
dumpPython(): remove wrong-type unused parameter

peruzzim committed:
Always use HLT process name to get trigger objects

fwyzard committed:
dumpPython(): remove wrong-type unused parameter

fwyzard committed:
dumpPython(): remove wrong-type unused parameter

peruzzim committed:
Merge pull request #438 from peruzzim/fixHLTname

Always use HLT process name to get trigger objects

Martin-Grunewald committed:
Some work on const-ness

Martin-Grunewald committed:
Code-Format

smorovic committed:
Per suggestion of fcossuti: move DaqTestHistograms.cc to DQM/HLTEvF
package

jaehyeok committed:
update mahidebugger to store all 8 pulses

jaehyeok committed:
update mahidebugger to store all 8 pulses

jaehyeok committed:
initialize nnlsWork_.nP at the beginning of phase1Apply

jaehyeok committed:
initialize nnlsWork_.nP at the beginning of phase1Apply

tocheng committed:
Add tau pog tags to heavy Ion GTs

Dr15Jones committed:
Explicitly use std::isinf in DeepCMVATagInfoProducer

The compilation under Centos 8 was failing without the explicit
use of the std namespace when calling isinf.

smuzaffar committed:
use python from env instead of /usr/bin/python

smuzaffar committed:
updated architecture to mach cc* too

Commit:
Take the comments

Commit:
Code check

Commit:
Correct wafer size and wafer positioning code

Commit:
Code check

Commit:
Avoid crash in making payloads

Commit:
Take out corrections in v9/v10 geometry

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28193/12294

- This PR adds an extra 24KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)
- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28193/12294/code-format.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28193/12294/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

jaehyeok committed:
code style update

jaehyeok committed:
code style update

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28193/12295

- This PR adds an extra 24KB to repository

cmsbuild commented:
A new Pull Request was created by  @jaehyeok  for master.

It involves the following packages:

RecoLocalCalo/HcalRecAlgos


@perrotta, @cmsbuild, @slava77 can you please review it and eventually sign? Thanks.
@apsallid, @mariadalfonso, @argiro this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

Martin-Grunewald committed:
Remove empty beginJob endJob methods

cmsbuild committed:
Merge pull request #28187 from Dr15Jones/fixRecoBTagCombined

Explicitly use std::isinf in DeepCMVATagInfoProducer

cmsbuild committed:
Merge pull request #28189 from smuzaffar/fix-slc-cc-arch

updated architecture to mach cc* too

cmsbuild commented:
+1
Tested at: d7faa423240ea268d63f227d3475598163924782
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-765864/3001/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild committed:
Merge pull request #28184 from fwyzard/fix_ParameterSet_PrintOptions

Fix usage of PrintOptions

cmsbuild committed:
Merge pull request #28188 from smuzaffar/fix-usr-bin-python

use python from env instead of /usr/bin/python

cmsbuild committed:
Merge pull request #28151 from smorovic/11X_parameterChangeAndTests

DAQ: updating module parameters and a new unit test

fabiocos committed:
Remove obsolete test configuration

davidlange6 committed:
clean cms tag collector / cvs services

peruzzim committed:
Merge pull request #435 from peruzzim/fixErasNanoUL

Fix usage of eras for UL samples

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-765864/3001/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 15718 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2961064
 - DQMHistoTests: Total failures: 9399
 - DQMHistoTests: Total nulls: 20
 - DQMHistoTests: Total successes: 2951304
 - DQMHistoTests: Total skipped: 341
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.015 KiB( 33 files compared)
 - DQMHistoSizes: changed ( 136.85,... ): 0.008 KiB        JetMET/SUSYDQM
 - DQMHistoSizes: changed ( 4.53 ): -0.008 KiB       JetMET/SUSYDQM
 - DQMHistoSizes: changed ( 136.788 ): -0.027 KiB       JetMET/SUSYDQM
 - DQMHistoSizes: changed ( 1000.0 ): 0.004 KiB        JetMET/SUSYDQM
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

guitargeek committed:
clean redundant member functions from ElectronSeed

fabiocos committed:
Fix use of TimeType enum after PR 28186

guitargeek committed:
code-format

fabiocos committed:
update to conddblib.py as suggested by S. Muzaffar

guitargeek committed:
don't implement ElectronSeedProducer config twice

guitargeek committed:
slight cleanup of ElectronSeedProducer config

schneiml committed:
Fix undersized buffers.

GCC 8.3.1 with -g -Og reported these and failed on them due to -Werror.

Dr15Jones committed:
Modernized XMLIdealGeometryESProducer

-specify consumes and use ESGetToken
-added fillDescriptions

slava77 commented on RecoLocalCalo/HcalRecAlgos/src/MahiFit.cc:
```diff
@@ -548,14 +546,46 @@ void MahiFit::phase1Debug(const HBHEChannelInfo& channelData, MahiDebugInfo& mdi
+        mdi.n1Pulse[iTS] = nnlsWork_.pulseMat.col(iBX).coeff(iTS);
+      }
+    } else if (nnlsWork_.bxs.coeff(iBX) == -2) {
+      mdi.p2Energy = 0;
```
this pattern seems to be a good case to replace n[1-4]\* and p[1-3]\* with an array of size 7 so that all of the code can fit in 5-7 lines instead of 40

jaehyeok replied:
Replaced n[1-4]* and p[1-3]* with **ootEnergy[7]** and **ootPulse[7][MaxSVSize]**.

slava77 commented on RecoLocalCalo/HcalRecAlgos/test/MahiDebugger.cc:
```diff
@@ -158,16 +158,26 @@ class MahiDebugger : public edm::one::EDAnalyzer<edm::one::SharedResources> {
-  float nEnergy;    //SOI+1 charge
+  float p1Energy;   //SOI-1 charge
+  float p2Energy;   //SOI-2 charge
+  float p3Energy;   //SOI-3 charge
```
an appropriate array in this file seems like a better replacement to an explicitly unrolled sequence of calls.

jaehyeok replied:
These have been replaced with an array **ootEnergy[7]**.

slava77 commented on RecoLocalCalo/HcalRecAlgos/src/MahiFit.cc:
```diff
@@ -548,14 +546,46 @@ void MahiFit::phase1Debug(const HBHEChannelInfo& channelData, MahiDebugInfo& mdi
     } else if (nnlsWork_.bxs.coeff(iBX) == -1) {
-      mdi.pEnergy = nnlsWork_.ampVec.coeff(iBX);
+      mdi.p1Energy = 0;
+      mdi.p1Energy = nnlsWork_.ampVec.coeff(iBX);
```
changes here seem to go beyond the PR description, which mentions only "Bug fix in initializing nP variable".
Please check and modify the PR description appropriately.

slava77 replied:
ah, nm, this is in the ```phase1Debug``` method.

fabiocos committed:
Initial cleaning of obsolete stuff in Validation/Performance

fabiocos committed:
Remove unused seeds fragments, move digi dumper into SimGeneral/Debugging as more appropriate location

Commit:
Attempt to remove access of DDCompactView from SimG4CMS/Calo code (Step 2)

Commit:
Code Check

guitargeek committed:
Move GsfElectronAlgoHeavyObjectCache together with GsfElectronAlgo

cmsbuild committed:
Merge pull request #28194 from fabiocos/fc-cleanvalid

Remove obsolete Validation/RecoEgamma test configuration

cmsbuild committed:
Merge pull request #28197 from fabiocos/fc-fxconddb

Fix use of TimeType enum after PR 28186

cmsbuild committed:
Merge pull request #28167 from ianna/dd4hep-ddddetector-fromdb-v1

DD4hep: Add DDDetector Producer from DB and Cleanup ES Records

hqucms committed:
Remove unused argument.

Commit:
Added inverse beta variable+error to pat::Muon

cmsbuild committed:
Merge pull request #28192 from bsunanda/Run3-fwk03

Run3-fwk03 Avoid crash in making payloads

cmsbuild committed:
Merge pull request #28198 from guitargeek/ElectronSeed_duplicate_functions

Clean redundant member functions from ElectronSeed

cmsbuild committed:
Merge pull request #28128 from hqucms/deeptau-batch-master

Speed up DeepTau by running the inner/outer convolution in batch

Commit:
A fix for missing SelectEvents configuration option

davidlange6 committed:
add missing dependencies

ianna committed:
Register shapes for delayed processing

ianna committed:
code check fix

ianna committed:
code format fixes

Dr15Jones committed:
Added consumes usage to HcalHardcodeCalibrations

This included use of ESGetTokens.

Dr15Jones committed:
Added consumes calls to MTDTransientTrackingRecHitBuilderESProducer

This includes using edm::ESGetToken.

cmsbuild committed:
Merge pull request #28134 from tocheng/TagsForTauPOG_110X

Add tau pog tags to GTs

cmsbuild committed:
Merge pull request #28173 from makortel/edmConfigDumpPrune

Make Process.prune() to prune also Tasks, and add --prune option to edmConfigDump

cmsbuild committed:
Merge pull request #28196 from cms-nanoAOD/pr_nanov6_part3_master

NanoAODv6, config fixes for UL

ianna committed:
Update unit tests

ianna committed:
Keep different root

cmsbuild committed:
Merge pull request #28211 from ianna/dd4hep-ddddetector-fromdb-v2

DD4Hep: DDDetector from DB - allow to delay a Solid Definition

cmsbuild committed:
Merge pull request #28120 from Dr15Jones/mayConsumeVolumeBasedMagneticFieldESProducerFromDB

Use mayConsumes in VolumeBasedMagneticFieldESProducerFromDB

peruzzim committed:
Fix wrong usage of associatedPackedPFCandidates vector in MuonFSRProducer

Martin-Grunewald committed:
Use global instead of stream modules

jaehyeok committed:
use arrays instead of unrolled sequence of calls

jaehyeok committed:
use arrays instead of unrolled sequence of calls

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28193/12350

- This PR adds an extra 28KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)
- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28193/12350/code-format.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28193/12350/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

jaehyeok committed:
code checks

jaehyeok committed:
code checks

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28193/12351

- This PR adds an extra 28KB to repository

cmsbuild commented:
Pull request #28193 was updated. @perrotta, @cmsbuild, @slava77 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

davidlange6 committed:
missing includes/dependencies

Commit:
Code-checks compliance fixes

Commit:
Code check compliance fixes for the header file

cmsbuild commented:
+1
Tested at: 1ae73ca02b4449e75c00924982fb9fd7f39a0f6c
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-765864/3049/summary.html

cmsbuild commented:
Comparison job queued.

ianna committed:
Update to the new dd4hep version

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-765864/3049/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 15708 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2961064
 - DQMHistoTests: Total failures: 9399
 - DQMHistoTests: Total nulls: 20
 - DQMHistoTests: Total successes: 2951304
 - DQMHistoTests: Total skipped: 341
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.015 KiB( 33 files compared)
 - DQMHistoSizes: changed ( 1000.0 ): 0.004 KiB        JetMET/SUSYDQM
 - DQMHistoSizes: changed ( 4.53 ): -0.008 KiB       JetMET/SUSYDQM
 - DQMHistoSizes: changed ( 136.731,... ): 0.008 KiB        JetMET/SUSYDQM
 - DQMHistoSizes: changed ( 136.788 ): -0.027 KiB       JetMET/SUSYDQM
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

cmsbuild committed:
Merge pull request #28175 from cms-tsg-storm/HLTDisplacedmumumuVtxProducer110X

Make HLTDisplacedmumumuVtxProducer a stream producer

cmsbuild committed:
Merge pull request #28208 from andrius-k/dqm-fix-SelectEvents-EVD-2instance

A fix for missing SelectEvents configuration option

cmsbuild committed:
Merge pull request #28209 from davidlange6/dl191118

add missing dependencies in DataFormats BuildFiles

cmsbuild committed:
Merge pull request #28217 from Dr15Jones/consumesHcalHardcodeCalibrations

Added consumes usage to HcalHardcodeCalibrations

cmsbuild committed:
Merge pull request #28171 from bsunanda/Phase2-hgx224

Phase2-hgx224 Select channels affected by mousebite

cmsbuild committed:
Merge pull request #28204 from fabiocos/fc-validperf

Clean Validation/Performance

Dr15Jones committed:
Add consumes to CTPPSInterpolatedOpticalFunctionsESSource

Dr15Jones committed:
Use ESGetToken in MTDTopologyEP

Dr15Jones committed:
Added consumes to MuonNumberingInitialization

Also use the ESGetToken obtained.

Dr15Jones committed:
Added consume for TrackerParametersESModule

Use the ESGetToken that is provided.

makortel committed:
Modernize EventWithHistoryEDFilter

- Move to global
- Remove empty methods
- Use range-for
- Use trailing underscore for member data following the code guidelines

wddgit committed:
Bug fix for concurrent IOVs

Only affects corner case with nConcurrentIOVs set to
more than one and the appendToDataLabel parameter set
to a non-empty value. The symptom is an assert failure,
when the assert should pass. The assert check didn't take
into account that the parameter could change the DataKey
labels.

Dr15Jones committed:
Added consumes to modules in DetectorDescription/DDCMS

Make use of the ESGetToken obtained.

cmsbuild committed:
Merge pull request #28218 from Dr15Jones/consumesMTDTransientTrackingRecHitBuilderESProducer

Added consumes calls to MTDTransientTrackingRecHitBuilderESProducer

guitargeek committed:
Remove obsolete HoE configuration variables in ElectronSeedProducer

cmsbuild committed:
Merge pull request #28221 from peruzzim/fixEleCandFSR

Fix electron PFCandidates RefVector usage in MuonFSRProducer

Commit:
Correct wafer size for the HGCal V11 version

Commit:
Also have the correct wafer size for v9 & v10

Commit:
Correct the xml files for October TB simulation

makortel committed:
Make debu_ member const

Co-Authored-By: Chris Jones <chrisdjones15@gmail.com>

Dr15Jones committed:
Fix gcc 9 warnings in SimMuon/MCTruth

Make sure all elements of a struct are set.

davidlange6 committed:
code patch

Dr15Jones committed:
Add fallthrough to LinkByRecHit

This fixes a warning in gcc9.

cmsbuild committed:
Merge pull request #28230 from wddgit/bugFixConcurrentIOVs

Bug fix for concurrent IOVs

cmsbuild committed:
Merge pull request #28038 from schneiml/dqm-reduce-api

DQM: remove support for References inside CMSSW

kpedro88 committed:
include dedx producers for fastsim

cmsbuild committed:
Merge pull request #28224 from ianna/dd4hep-update-version-and-tests-v1

DD4Hep: Update to the new DD4hep version

Commit:
Remove some of the checks in the trigger DetId constructore

kpedro88 committed:
remove unwanted check

Commit:
Code check

cmsbuild committed:
Merge pull request #28116 from steggema/TICLtoPF

TICL: Create PF candidates as output of TICL

cmsbuild committed:
Merge pull request #28191 from bsunanda/Phase2-hgx225

Phase2-hgx225 Correct wafer positioning code

Commit:
Removed unused inverseBeta setters in pat::Muon

Some additional minor naming changes in the muon extra timer info reader

ianna committed:
Delay referenced materials resolution

ianna committed:
code format

cmsbuild committed:
Merge pull request #28213 from guitargeek/ElectronSeedProducer_config

Remove manually generated cfi file for ElectronSeedProducer

cmsbuild committed:
Merge pull request #28099 from schneiml/dqm-new-onemodules

DQM: Restructure the edm::one DQMEDAnalyzer base classes

cmsbuild committed:
Merge pull request #28202 from schneiml/cmssw-fix-undersized-buffers

Fix undersized buffers.

cmsbuild committed:
Merge pull request #28215 from Dr15Jones/consumesXMLIdealGeometryESProducer

Modernized XMLIdealGeometryESProducer

cmsbuild committed:
Merge pull request #28222 from davidlange6/dl191021

a few missing includes/dependencies in DataFormats

cmsbuild committed:
Merge pull request #28226 from Dr15Jones/consumesCTPPSInterpolatedOpticalFunctionsESSource

Add consumes to CTPPSInterpolatedOpticalFunctionsESSource

cmsbuild committed:
Merge pull request #28227 from Dr15Jones/consumesMTDTopologyEP

Use ESGetToken in MTDTopologyEP

cmsbuild committed:
Merge pull request #28228 from Dr15Jones/consumesTrackerParametersESModule

Added consume for TrackerParametersESModule

cmsbuild committed:
Merge pull request #28229 from Dr15Jones/consumesMuonNumberingInitialization

Added consumes to MuonNumberingInitialization

cmsbuild committed:
Merge pull request #28231 from makortel/modernizeEventWithHistoryEDFilter

Modernize EventWithHistoryEDFilter

cmsbuild committed:
Merge pull request #28232 from Dr15Jones/consumesDetectorDescriptionDDCMS

Added consumes to modules in DetectorDescription/DDCMS

kpedro88 committed:
simplify tasks

cmsbuild committed:
Merge pull request #28239 from Dr15Jones/gcc9SimMuonMCTruth

Fix gcc 9 warnings in SimMuon/MCTruth

cmsbuild committed:
Merge pull request #28240 from Dr15Jones/fallthroughLinkByRecHit

Add fallthrough to LinkByRecHit

cmsbuild committed:
Merge pull request #28249 from bsunanda/Run3-hcx234

Run3-hcx234 Remove some of the checks in the trigger DetId constructore

Commit:
Moved clarification comment to variable getter instead of the generic Muon Extra Timer reader

smuzaffar committed:
python3 filter return iterator instead of list

smuzaffar committed:
[GCC9] fix implicit-fallthrough warning for Validation/RecoParticleFlow

smuzaffar committed:
[GCC9] fix implicit-fallthrough warning for Validation/EcalDigis

Commit:
Fix gcc9 warning in DetectorDescription/DDCMS (missing struct initializer)

smuzaffar committed:
[GCC9] fix base class initialization warning for RecoTracker/TkHitPairs

smuzaffar committed:
[GCC9] fix left shift of negative value warning for SimRomanPot/SimFP420

smuzaffar committed:
[GCC9] fixed missing initializer for member warning for RecoParticleFlow/PFClusterProducer

smuzaffar committed:
[GCC9] Fix for empty else body in L1TriggerConfig/L1GtConfigProducers

Commit:
Update DetectorDescription/DDCMS/interface/Filter.h

compiles without warning with 9 as well

Co-Authored-By: Chris Jones <chrisdjones15@gmail.com>

cmsbuild committed:
Merge pull request #28273 from smuzaffar/gcc9-RecoecoTrackerTkHitPairs

[GCC9] fix base class initialization warning for RecoTracker/TkHitPairs

cmsbuild committed:
Merge pull request #28276 from smuzaffar/gcc9-ecoParticleFlowPFClusterProducer

[GCC9] fixed missing initializer for member warning for PFClusterProducer

cmsbuild committed:
Merge pull request #28278 from smuzaffar/gcc9-L1TriggerConfigL1GtConfigProducers

[GCC9] Fix for empty else body in L1TriggerConfig/L1GtConfigProducers

cmsbuild committed:
Merge pull request #28237 from bsunanda/Run3-TB45

Run3-TB45 Correct the xml files for October TB simulation

cmsbuild committed:
Merge pull request #28195 from davidlange6/dl191017

clean cms tag collector / cvs services

smuzaffar committed:
Merge pull request #28266 from cms-sw/smuzaffar-patch-2

[PY3] Fix needed for O2O unit tests

cmsbuild committed:
Merge pull request #28205 from bsunanda/Run3-hcx232

Run3-hcx232 Attempt to remove access of DDCompactView from SimG4CMS/Calo code (Step 2)

cmsbuild committed:
Merge pull request #28212 from cericeci/reco_include_InverseBeta

Propagate inverse beta variable to pat muons (muon POG)

cmsbuild committed:
Merge pull request #28236 from bsunanda/Phase2-hgx226

Phase2-hgx226 Correct wafer size for the HGCal V11 version

cmsbuild committed:
Merge pull request #28257 from kpedro88/FastDeDxContent_1100pre10

Include dedx collections in fastsim output

cmsbuild committed:
Merge pull request #28270 from mrodozov/fix-gcc9-warns-DetectorDescription/DDCMS

Fix gcc9 warning in DetectorDescription/DDCMS (missing struct initial…

mariadalfonso commented:
@slava77 would be great to have this merged in the pre11.
is there some explanation needed from our side ?

ianna committed:
Cleanup

makortel committed:
Add attribute to denote intentional fallthrough

slava77 commented on RecoLocalCalo/HcalRecAlgos/src/MahiFit.cc:
```diff
@@ -546,16 +544,18 @@ void MahiFit::phase1Debug(const HBHEChannelInfo& channelData, MahiDebugInfo& mdi
       }
     } else if (nnlsWork_.bxs.coeff(iBX) == pedestalBX_) {
+      mdi.pedEnergy = 0;
       mdi.pedEnergy = nnlsWork_.ampVec.coeff(iBX);
```
L547 is unnecessary. 
Was it intentional, or was something else supposed to be assigned to 0?

jaehyeok replied:
L547 is removed.

jaehyeok replied:
L547 is removed

slava77 commented on RecoLocalCalo/HcalRecAlgos/src/MahiFit.cc:
```diff
@@ -546,16 +544,18 @@ void MahiFit::phase1Debug(const HBHEChannelInfo& channelData, MahiDebugInfo& mdi
+      else
+        ootIndex += 3;
+      mdi.ootEnergy[ootIndex] = 0;
+      mdi.ootEnergy[ootIndex] = nnlsWork_.ampVec.coeff(iBX);
```
similarly here, what is the purpose of L555?

jaehyeok replied:
L555 is removed.

jaehyeok replied:
L555 is removed.

slava77 commented on RecoLocalCalo/HcalRecAlgos/src/MahiFit.cc:
```diff
@@ -546,16 +544,18 @@ void MahiFit::phase1Debug(const HBHEChannelInfo& channelData, MahiDebugInfo& mdi
+        ootIndex += 2;
+      else
+        ootIndex += 3;
+      mdi.ootEnergy[ootIndex] = 0;
```
shouldn't there be a check or assert here that ootIndex is in the range from 0 to 6 (it can also be checked on L549)

Also, is the math above correct? If I got it right we have here 0 -> 3 and 1 -> 3. It may be OK if 0 is not possible.

jaehyeok replied:
Now restricted the range to be -3 to 4 in L549. 
```
    } else if (nnlsWork_.bxs.coeff(iBX)>=-3 && nnlsWork_.bxs.coeff(iBX) <= 4) {
```

Math is correct because 0 is not possible: 
``` 
    if (nnlsWork_.bxs.coeff(iBX) == 0) {
   ...
    } else if (nnlsWork_.bxs.coeff(iBX) == pedestalBX_) {
   ...
    } else if (nnlsWork_.bxs.coeff(iBX)>=-3 && nnlsWork_.bxs.coeff(iBX) <= 4) {
```

slava77 reviewed (commented):
sorry for the delay

cmsbuild committed:
Merge pull request #28260 from ianna/dd4hep-resolve-materials-v1

DD4hep: Referenced Material Resolution and Enabled all Tracker Algorithms

cmsbuild committed:
Merge pull request #28267 from smuzaffar/gcc9-ValidationRecoParticleFlow

[GCC9] fix implicit-fallthrough warning for Validation/RecoParticleFlow

cmsbuild committed:
Merge pull request #28268 from smuzaffar/gcc9-ValidationEcalDigis

[GCC9] fix implicit-fallthrough warning for Validation/EcalDigis

cmsbuild committed:
Merge pull request #28275 from smuzaffar/gcc9-SimRomanPotSimFP420

[GCC9] fix left shift of negative value warning for SimRomanPot/SimFP420

cmsbuild committed:
Merge pull request #28296 from makortel/fixGcc9NavigationSchoolAnalyzer

Fix GCC9 warning in NavigationSchoolAnalyzer

jaehyeok committed:
remove unnecessary initializations

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
Pull request #28193 was updated. @SiewYan, @andrius-k, @lveldere, @chayanit, @emeschi, @sbein, @schneiml, @ianna, @kpedro88, @Martin-Grunewald, @rekovic, @fioriNTU, @tlampen, @alberto-sanchez, @pohsun, @santocch, @peruzzim, @perrotta, @civanch, @zhenhu, @cmsbuild, @agrohsje, @fwyzard, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @ssekmen, @mdhildreth, @jfernan2, @tocheng, @slava77, @ggovi, @qliphy, @fabiocos, @mkirsano, @benkrikler, @efeyazgan, @kmaeshima, @christopheralanwest, @pgunnell, @franzoni, @fgolf, @mommsen can you please check and sign again.

abdoulline commented:
Jae, something went wrong, as too many commits have been pulled into your branch...

abdoulline commented:
I'm afraid you may need to close this PR and reopen another (new) one.

kpedro88 commented:
@jaehyeok you should never use `cms-merge-topic` in a PR. Please follow the guide here: http://cms-sw.github.io/tutorial-resolve-conflicts.html

kpedro88 commented:
-1

jaehyeok committed:
allow pulses only in -3 to 4

cmsbuild commented:
The code-checks are being triggered in jenkins.

kpedro88 commented:
-1
still broken

jaehyeok commented:
Sorry for the mess. I've opened a new PR: https://github.com/cms-sw/cmssw/pull/28309.

perrotta commented:
@jaehyeok , please close this PR then

perrotta commented:
-1
(moved to  #28309)

jaehyeok commented:
This PR has moved to https://github.com/cms-sw/cmssw/pull/28309, so will be closed.
